vm/src/builtins/make_module.rs

```diff
@@ -292,30 +321,17 @@ mod decl {
             })
     }
 
-    fn catch_attr_exception<T>(
-        ex: PyBaseExceptionRef,
-        default: T,
-        vm: &VirtualMachine,
-    ) -> PyResult<T> {
-        if ex.isinstance(&vm.ctx.exceptions.attribute_error) {
-            Ok(default)
-        } else {
-            Err(ex)
-        }
-    }
-
     #[pyfunction]
     fn getattr(
         obj: PyObjectRef,
         attr: PyStrRef,
         default: OptionalArg<PyObjectRef>,
         vm: &VirtualMachine,
     ) -> PyResult {
-        let ret = vm.get_attribute(obj, attr);
         if let OptionalArg::Present(default) = default {
-            ret.or_else(|ex| catch_attr_exception(ex, default, vm))
+            Ok(vm.get_attribute_opt(obj, attr)?.unwrap_or(default))
```

Comment:
is this better to be changed like this?

Reply:
I think so, I added the get_attribute_opt function for use in _ast, but I realized that the logic was being duplicated here as well.